src/memory/AmdFlash.cc

```diff
@@ -186,17 +191,15 @@ AmdFlash::~AmdFlash() = default;
 AmdFlash::GetSectorInfoResult AmdFlash::getSectorInfo(size_t address) const
 {
 	address &= size() - 1;
-	auto it = sectorInfo.begin();
+	auto it = chip.geometry.regions.begin();
 	size_t sector = 0;
-	while (address >= it->size) {
-		address -= it->size;
-		++sector;
+	while (address >= (it->count << it->size.exponent)) {
+		address -= it->count << it->size.exponent;
+		sector += it->count;
 		++it;
-		assert(it != sectorInfo.end());
+		assert(it != chip.geometry.regions.end());
 	}
-	auto sectorSize = it->size;
-	auto offset = address;
-	return {sector, sectorSize, offset};
+	return {sector + (address >> it->size.exponent), it->size(), address & (it->size() - 1)};
```

Comment:
Similar as above. For `>>` we could add an `operator/` in `power_of_two`.

Reply:
Will do, also operator% for the `& size-1`.